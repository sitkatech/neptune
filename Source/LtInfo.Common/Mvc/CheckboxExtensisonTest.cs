/*-----------------------------------------------------------------------
<copyright file="CheckboxExtensisonTest.cs" company="Sitka Technology Group">
Copyright (c) Sitka Technology Group. All rights reserved.
<author>Sitka Technology Group</author>
</copyright>

<license>
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License <http://www.gnu.org/licenses/> for more details.

Source code is available upon request via <support@sitkatech.com>.
</license>
-----------------------------------------------------------------------*/
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using List = DocumentFormat.OpenXml.Office2010.ExcelAc.List;

namespace System.Web.Mvc.Html
{
    [TestFixture]
    public class CheckboxExtensisonTest
    {

        private readonly List<SelectListItem> _testSelectList = new List<SelectListItem>()
        {
            new SelectListItem() {Value = "1_testValue", Text = "someText_1"},
            new SelectListItem() {Value = "2_testValue", Text = "someText_2"},
            new SelectListItem() {Value = "3_testValue", Text = "someText_3"},
            new SelectListItem() {Value = "4_testValue", Text = "someText_4"}
        };

        private readonly List<SelectListItem> _testSelectList2 = new List<SelectListItem>()
        {
            new SelectListItem() {Value = "5_testValue", Text = "someText_5"},
            new SelectListItem() {Value = "6_testValue", Text = "someText_6"},
            new SelectListItem() {Value = "7_testValue", Text = "someText_7"},
            new SelectListItem() {Value = "8_testValue", Text = "someText_8"},
            new SelectListItem() {Value = "9_testValue", Text = "someText_9"}
        };
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void ListItemToCheckboxTest()
        {
            var result = CheckboxExtensions.ListItemToCheckbox(new SelectListItem() {Value = "someValue", Text = "someText"}, new Dictionary<string, object>(), "someName");
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void BuildItemsTest()
        {
            var result = CheckboxExtensions.BuildItems("testOptionLabel", _testSelectList, new Dictionary<string, object>(), "testFullName");
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void BuildItemsTwoColumnsTest()
        {
            var result = CheckboxExtensions.BuildItems("testOptionLabel", _testSelectList, new Dictionary<string, object>(), "testFullName",CheckboxExtensions.ColNumber.twoCols);
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void BuildItemsTwoColumnsTest_2()
        {
            var result = CheckboxExtensions.BuildItems("testOptionLabel", _testSelectList2, new Dictionary<string, object>(), "testFullName", CheckboxExtensions.ColNumber.twoCols);
            Approvals.Verify(result);
        }
    }
}
