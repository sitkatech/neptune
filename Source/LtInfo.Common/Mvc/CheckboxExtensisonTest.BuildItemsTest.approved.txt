<label for="testFullName"><input id="testFullName" name="testFullName" type="checkbox" value=""></input> testOptionLabel</label>
<label for="testFullName1_testValue"><input id="testFullName1_testValue" name="testFullName" type="checkbox" value="1_testValue"></input> someText_1</label>
<label for="testFullName2_testValue"><input id="testFullName2_testValue" name="testFullName" type="checkbox" value="2_testValue"></input> someText_2</label>
<label for="testFullName3_testValue"><input id="testFullName3_testValue" name="testFullName" type="checkbox" value="3_testValue"></input> someText_3</label>
<label for="testFullName4_testValue"><input id="testFullName4_testValue" name="testFullName" type="checkbox" value="4_testValue"></input> someText_4</label>
